firefox-ios/Client/Frontend/Home/LegacyHomepageViewController.swift

```diff
@@ -162,6 +161,8 @@ class LegacyHomepageViewController:
 
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
+        // FXIOS-9428 - Record telemetry in viewDidAppear since viewWillAppear is sometimes triggered twice
+        viewModel.recordViewAppeared()
```

Comment:
The method still has the microsurvey event call, so the issue still persists since it's being called in `viewDidAppear` now. What about moving out the following code back into `viewWillAppear`? 

```
viewWillAppearEventThrottler.throttle {
    Experiments.events.recordEvent(BehavioralTargetingEvent.homepageViewed)
}
```

To test the issue with the microsurvey trigger, here are the following steps:

- Fresh install the app + confirm that you have viewed the homepage at least once
- Tap on the private mask toggle (which will bring you to private homepage) and tap on it again (which will bring you to normal homepage)
- Verify that the microsurvey appears on the 2nd view of normal homepage

In the original PR, the microsurvey didn't appear until after navigating elsewhere. While on main it would appear as soon as the 2nd view of the normal homepage appears.

**Issue**

https://github.com/user-attachments/assets/feaf7983-dd29-4d3a-a106-900abc1ea7cb

**Expected**

https://github.com/user-attachments/assets/6cdb85bd-01d6-4be1-bbb3-cfd34df13fa1

Reply:
Thank you Cyndi! I moved back the code and see the expected behavior for the microsurvey.